runelite-client/src/main/java/net/runelite/client/plugins/runenergy/RunEnergyPlugin.java

```diff
@@ -199,30 +223,45 @@ String getEstimatedRunTimeRemaining(boolean inSeconds)
 		}
 	}
 
-	private boolean isLocalPlayerWearingFullGraceful()
+	private int getGracefulRecoveryBoost()
 	{
 		final ItemContainer equipment = client.getItemContainer(InventoryID.EQUIPMENT);
 
 		if (equipment == null)
 		{
-			return false;
+			return 0;
 		}
 
 		final Item[] items = equipment.getItems();
 
-		// Check that the local player is wearing enough items to be using full Graceful
-		// (the Graceful boots will have the highest slot index in the worn set).
-		if (items == null || items.length <= EquipmentInventorySlot.BOOTS.getSlotIdx())
+		if (items == null)
+		{
+			return 0;
+		}
+
+		int boost = 0;
+
+		for (final GracefulEquipmentSlot slot : GracefulEquipmentSlot.values())
 		{
-			return false;
+			if (items.length <= slot.getIndex())
+			{
+				continue;
+			}
+
+			final int wornItemId = items[slot.getIndex()].getId();
```

Comment:
Entries are not guaranteed to be nonnull. Change this to `final Item wornItem` and null-check it along with checking `slot.getItems().contains(wornItem.getId())` below.